jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -105,6 +105,9 @@ def _dataset_selected_changed(self, event={}):
             if telescope == 'JWST':
                 if 'photometry' in meta and 'pixelarea_arcsecsq' in meta['photometry']:
                     self.pixel_area = meta['photometry']['pixelarea_arcsecsq']
+                if 'bunit_data' in meta and meta['bunit_data'] == u.Unit("MJy/sr"):
+                    # Hardcode the flux conversion factor from MJy to ABmag
+                    self.flux_scaling = 3631

```

Comment:
Since we're talking about MJy and not Jy, the mega needs to be factored out as well.
```suggestion
                    self.flux_scaling = 3631e-6
```

```python
>>> import numpy as np
>>> from astropy import units as u
>>> (1 * u.Jy).to(u.ABmag)
<Magnitude 8.9 mag(AB)>
>>> -2.5 * np.log10(1 / 3631)  # How the flux_scaling is used
8.90006562228223
>>> (1 * u.MJy).to(u.ABmag)
<Magnitude -6.1 mag(AB)>
>>> -2.5 * np.log10(1 / 3631e-6)
-6.099934377717769
```